test/perf/dotnet-tools/Reporting/Reporter.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+
+namespace Reporting
+{
+    public class Reporter
+    {
+        private Run run;
+        private Os os;
+        private Build build;
+        private List<Test> tests = new List<Test>();
+        protected IEnvironment environment;
+
+        private Reporter() { }
+
+        public void AddTest(Test test)
+        {
+            if (tests.Any(t => t.Name.Equals(test.Name)))
+                throw new Exception($"Duplicate test name, {test.Name}");
+            tests.Add(test);
+        }
+
+        /// <summary>
+        /// Get a Reporter. Relies on environment variables.
+        /// </summary>
+        /// <param name="environment">Optional environment variable provider</param>
+        /// <returns>A Reporter instance or null if the environment is incorrect.</returns>
+        public static Reporter CreateReporter(IEnvironment environment = null)
+        {
+            var ret = new Reporter();
+            ret.environment = environment == null ? new EnvironmentProvider() : environment;
+            if (ret.InLab)
+            {
+                ret.Init();
+            }
+                        
+            return ret;
+        }
+
+        private void Init()
+        {
+            run = new Run
+            {
+                CorrelationId = environment.GetEnvironmentVariable("HELIX_CORRELATION_ID"),
+                PerfRepoHash = environment.GetEnvironmentVariable("PERFLAB_PERFHASH"),
+                Name = environment.GetEnvironmentVariable("PERFLAB_RUNNAME"),
+                Queue = environment.GetEnvironmentVariable("PERFLAB_QUEUE"),
```

Comment:
We don't have a hardware lab, so it's definitely a preferable option for us if it's possible for us to run our perf tests in the .NET performance team hardware lab 😄. Even if we cannot use the .NET team lab, maybe it's still preferable for us to keep the same format of the exported reports, so that we can leverage the tools from the .NET performance teams easily?